tests/python/relax/test_dataflow_pattern.py

```diff
@@ -1034,15 +1034,6 @@ def main(
         matmul2 = is_op("relax.matmul")(inp_pat, K_weight_pat)
         matmul3 = is_op("relax.matmul")(inp_pat, V_weight_pat)
 
-        # TODO(masahi): Automate addition of used_by constraints during is_op
-        inp_pat.used_by(matmul1, 0)
-        inp_pat.used_by(matmul2, 0)
-        inp_pat.used_by(matmul3, 0)
-
-        Q_weight_pat.only_used_by(matmul1, 1)
-        K_weight_pat.only_used_by(matmul2, 1)
-        V_weight_pat.only_used_by(matmul3, 1)
```

Comment:
I understand `used_by` can be removed now as it is implied by a function call. However, `only_used_by` is a stronger constraint that cannot be implied via a fn call right?

Reply:
But yeah I guess you meant oftentimes we don't fold Q, K, and V so it is fine if they are being used by others at the same time.